### PR TITLE
chore: remove the limited support there was for Windows

### PR DESCRIPTION
**Pull request by jansegre**
### Motivation

Windows support has been very bare, undertested and there are no use cases that could not run on WSL instead.

### Acceptance Criteria

- Remove CI jobs that run on Windows
- Remove specific dependencies and markers
- Remove tests workarounds

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

File: pyproject.toml
Comment (glevco) [inline pyproject.toml:1]: Hijacking a random file just to open a thread, you missed a few places, such as `reactor.py`. Grepping `win32` should be enough

**Review by glevco — APPROVED** [review]

**Review by msbrogli — DISMISSED** [review]

File: pyproject.toml
Comment (jansegre) [reply pyproject.toml:1]: Oh, I only grepped `win32` on the tests, I'll add those too.

Comment (jansegre) [reply pyproject.toml:1]: Done.

**Review by glevco — APPROVED** [review]

File: pyproject.toml
Comment (glevco) [reply pyproject.toml:1]: Thanks!

**Review by msbrogli — APPROVED** [review]

**Pull request merged**